### PR TITLE
Research Warcrimes

### PR DESCRIPTION
**Pull request by Unknownity**
## About The Pull Request

This the long awaited Research update that has been in development hell for 9 months. I'll post the words of the creator himself of what's it about and ofcourse I will say some words myself.

This MR is intended to give Research non-medical stuff to do in addition to medical stuff, and fix a few issues that were brought up with custom reagents as a whole.
So firstly, this adds the capability for properties to have reaction_mob, reaction_turf, and reaction_obj procs. I've also added a bunch of these for existing and new properties. This is intended to make more interesting properties possible, as it opens up the possibility of using sprayers, chem grenades, and all those delicious, delicious warcrimes against xenos/hostile humans. OD grenades already existed, but weren't really that useful. In total, the properties I added/tweaked allow you to heal, stabilize, slow down/speed up, reduce healing, and reduce armor for xenos.
I also put fire-penetrating into the legendary properties pool, as a few people suggested it to me and it should still be very hard to get (especially with how few people actually use research fire chemicals or flamer fuel).
There are also some new effects for existing properties when they are over a certain level. The levels for triggering this correspond somewhat to the maximums for amplification/creation.
The cost of certain properties was also tweaked. In particular, I upped the cost of stimulant properties a bit, and reduced the costs of the organ-healing chems to make them easier to incorporate into things. This probably needs some review to make sure it doesn't unbalance anything too much.
I also changed how amplification and creation are limited. Instead of points being the main limiting factor, clearance now is (I originally used tech levels, but techwebs are disabled for now so instead I'm using clearance). This should hopefully be better for balance without gimping costs incredibly hard. I made amplification have a slightly higher level cap than creation, but if that's a bit too unbalanced I can change that. Point income also increases if Research buys clearance instead of having it swiped up, which should act as a sort of investment for points during long rounds.
The tweaks to chemical generation are meant to make recipes somewhat easier to make, and to make it so that you don't get 5 chemicals with the same 2 neutral/negative properties. This should also stop having the nutriment clones show up in every second/third chem you get. The reagents I chose to exclude are somewhat subject to change, though, depending on what people think should be included.
Finally, I also added a way to get Thwei that doesn't require vial boxes, and fixed a couple of bugs I found.

My note: I revived the MR back from the dead that has lasted for 9 months since BenBot has quit CM at some point and stopped maintaining the code, tested it and everything works as its supposed to from what I could tell, nothing broke on my end oh and since there were like what two seperate MRs before this which changed how research operated, one MR got effectively reverted, the one about recipes showing up, its now way better on how it works with the introduction of a new flag that prevent generation AND regular Nutriment can still appear in recipes (it would have stopped appearing in that MR). the other MR is about delta time introduction, most if not all reagents work on potency defines than delta time which is better (i guess, that's why he did it).
Transmission is still gone.

During back in gitlab era, the PR has been testmerged once or twice, no errors occured from the player's perspective with the only complaints being that some features are underpowered such as the spray not being a good chem injector and the extra passive credits per level (Both won't be buffed either for a good reason) Speaking about gitlab, here's the link to all the MR's related to this https://gitlab.com/cmdevs/colonial-warfare/-/merge_requests/2776 and https://gitlab.com/cmdevs/colonial-warfare/-/merge_requests/1587 if you need to look any further, the only issue left for the original MR was lack of maintaining it (rebasing etc.) and I fixed that. and the design doc https://github.com/cmss13-devs/cmss13-design-docs/blob/main/Research/ResearchWarcrimes.md

## Why It's Good For The Game

Provides Research to do more things than just stims or corrupted as the Trello says "Research is not meant to be a stim only producing factory."
Alongside with the IO PR, it will bring some life back to the department.

## Changelog

:cl: Benbot16
add: Some chemical properties now have effects when applied to mobs via touch. Some of these also effect xenos.

Adds components for speed adjustment and healing reduction.

Adds a few new properties using the above components.

Some properties now have additional effects at higher levels.

You can now get Thwei by grinding crystals in Industrial Grinders (instead of having to rely on vial box RNG).

balance: tweaks the cost of some properties

Tweaks chemical generation so that properties are (mostly) unique for CL 1/2 chemicals.

Prevents certain reagents (Such as antag_stimulant and the nutriment subtypes) from spawning in chemical recipes.

Levels in Amplify and Create mode for the Synthesis Simulator are now limited by the current clearance level. This amounts to a level max of 4(at CL1/2)/8(at CL3/4)/10(at CL5) for Create mode, and 4/6/8/10/unlimited for Amplify mode.

Buying clearance levels will now slightly increase passive research income.

fix: Passive research credit gain now works in Extended
/:cl:

### CODE REVIEW

**Comment by Watermelon914** [comment]
link your design doc in your description too

**Comment by Watermelon914** [comment]
approved design doc can be found here
https://github.com/cmss13-devs/cmss13-design-docs/blob/main/Research/ResearchWarcrimes.md

**Comment by Weuyn** [comment]
I like the new chemical generation code, definitely noticing more fun stuff and less useless nutritious/emetic/electrogenic junk. The synthesis machine nerf sucks though. Not the potencies part, thats irrelevant, but the passive points gain is so important post research nerf.

Like, it takes 20 points to make a 2 property chemical if you've mot been extremely lucky on plant chem RNG: A plant chemical needs 25-30u OD and a good recipe for it to be worth modifying. Chemicals with 5-20u are much more common and useless if they aren't good to distribute as-is. Wouldn't be the case if the duration multipliers on chems from 0.5x to 5x weren't removed with the godstims but they were.

When the average round is decided in under an hour, just buying more clearance for faster points gain is not an option either.

**Comment by Weuyn** [comment]
There is also frequently no ordnance tech or they are uninterested jn whatever chemical you have made, limiting your options for offensive chemicals to marines mixing one beaker into another, splashing a xeno with a beaker or using a spray bottle.

Need to be able to make our own grenades

**Comment by Unknownity** [comment]
> I like the new chemical generation code, definitely noticing more fun stuff and less useless nutritious/emetic/electrogenic junk. The synthesis machine nerf sucks though. Not the potencies part, thats irrelevant, but the passive points gain is so important post research nerf.
> 
> Like, it takes 20 points to make a 2 property chemical if you've mot been extremely lucky on plant chem RNG: A plant chemical needs 25-30u OD and a good recipe for it to be worth modifying. Chemicals with 5-20u are much more common and useless if they aren't good to distribute as-is. Wouldn't be the case if the duration multipliers on chems from 0.5x to 5x weren't removed with the godstims but they were.
> 
> When the average round is decided in under an hour, just buying more clearance for faster points gain is not an option either.

You should not be dependent on the increased passive bonus. It is what it is, a bonus. Most of the credit gain should come from finding new chemicals from hydro and planetside (the IO return PR will greatly help with the latter)

**Comment by Unknownity** [comment]
> There is also frequently no ordnance tech or they are uninterested jn whatever chemical you have made, limiting your options for offensive chemicals to marines mixing one beaker into another, splashing a xeno with a beaker or using a spray bottle.
> 
> Need to be able to make our own grenades

a player issue, MTs and the CE can substitute OTs if they're missing.
The only thing you're missing out from them is the capability of making grenades and mortar shells, while spray bottles are not as effective and are more dangerous due to the close contact required, some properties do benefit from it such as Toxic for weed removal

**Comment by Diegoflores31** [comment]
as a researcher player .. this PR while it fixes some anoying things like the removal of oxidizing and making thwei actually obtainable . it makes doing stims (main research feature rn) a complete pain and obliterates any chance of doing anything usefull before 1 or 2 hours into the round basically forcing you buy clearance 5 and pray for greenos .
while most medical properties has been unnerfed. the fact that you NEED clearance to make anything good is really awfull and the spray and ontouch feature i can 100% surely say it will be barely ever used and is straigh up useless as it is currently on this TM . 

sprays are not combat oriented . they are laggy and bad .
weed removal was already a thing with Plant B gone . and its not good.
sprays for "medicinal " uses are nonsense and you are better off with a injector or hypospray . 20 to 1 is a BAD ratio

if you could like fix fluorosulfatant maybe it could be a little better but appart from the thwei thing and the oxidising removal i dont see any improvement in research so far

**Comment by Weuyn** [comment]
> You should not be dependent on the increased passive bonus. It is what it is, a bonus. Most of the credit gain should come from finding new chemicals from hydro and planetside (the IO return PR will greatly help with the latter)
I'll have to try another round before the merge is removed, but there did not seem to be that many more chemicals in hydroponics, just the addition of one or two clearance 3's. IO will probably help though, and yeah too many points would mean 5X is too easy to get to and that unlocks all sorts of broken properties like hardening and nst and shit.

**Comment by Swendvideo** [comment]
Remember cloning? What if add it but make it very rare or legendary property, it will be a nice touch and it's fits research well

**Comment by Nanu308** [comment]
Numbers are from a look across a good starting point, i trust @stanalbatross to run required testing and follow up on this while i am away.

**Comment by RandomG05** [comment]
Been playing research for a while, so TrustMeBro™

Increasing credit income every clearance : Good, need this to do anything with the bought stuff

Capped property level when creating : this is great, more people should try making useful stuff out of what they already have.
The only downside is combined with shitty OD unit of randomly generated chems, thats 2 big limitations .

Disrupting, Neutralizing, etc : Look, if you're gonna add niche stuff with not much reward, and the only way to test or even utilize them is to cooperate with another department, yeah.. Either make it so the properties arent just a nuisance so its even worth creating/relating as a proper OT chem, give research engineering capabilities/something similiar, or give another method to utilize the chems (Tranquil guns, covering mags in chems, etc). 

Touch thingy : What? its cool and all but there's a reason doctors give us paracetamol pills and not paracetamol sprays.

Overall mechanics is alright, but for the components of so called Warcrimes is... a little off-target. Try putting crazy values for now and tweaking it down to target a balance. Also please make it so chems can be tested on animals/humans, how tf am i gonna get a live rooner, spray it with disrupting and ask it, bro how you feel? (ok i could)

**Comment by Unknownity** [comment]
> Been playing research for a while, so TrustMeBro™
> 
> Increasing credit income every clearance : Good, need this to do anything with the bought stuff
> 
> Capped property level when creating : this is great, more people should try making useful stuff out of what they already have. The only downside is combined with shitty OD unit of randomly generated chems, thats 2 big limitations .
> 
> Disrupting, Neutralizing, etc : Look, if you're gonna add niche stuff with not much reward, and the only way to test or even utilize them is to cooperate with another department, yeah.. Either make it so the properties arent just a nuisance so its even worth creating/relating as a proper OT chem, give research engineering capabilities/something similiar, or give another method to utilize the chems (Tranquil guns, covering mags in chems, etc).
> 
> Touch thingy : What? its cool and all but there's a reason doctors give us paracetamol pills and not paracetamol sprays.
> 
> Overall mechanics is alright, but for the components of so called Warcrimes is... a little off-target. Try putting crazy values for now and tweaking it down to target a balance. Also please make it so chems can be tested on animals/humans, how tf am i gonna get a live rooner, spray it with disrupting and ask it, bro how you feel? (ok i could)

I see where you're getting with this but for now, there is no consideration for Research to be able to make explosives on their own as it would be neglecting the OT department. When it comes to testing properties that have main effects for xenomorphs like disrupting, the property does have a negative effect if you use an injector on a human/monkey with it, other than that you just gotta believe the code/wiki since obviously most of the new additions can only be  "experienced" by one side.

If there will be constant reports of issues that there's nobody to assist with you through the OT side (no OTs,MTs,CE,Synth) then I will consider of an "emergency" solution that provides a machine capable of making you limited tool-assisted explosives that don't require skills to create.

**Comment by Segrain** [comment]
My availability comes and goes, but I will get to reviewing this mess one day.

**Comment by Weuyn** [comment]
can this get testmerged with the IO update? It hasn't ever beed so far as far as I know.

**Comment by Unknownity** [comment]
> can this get testmerged with the IO update? It hasn't ever beed so far as far as I know.

cannot be done until IO is fully merged due to conflicts between the PRs

**Comment by Diegoflores31** [comment]
Tested on TM. Its really really really hard to get anything done before 1:20 into the round winout the IO update . The only things we were able to acomplist lately was either ultrazine farms or roundstart chems that were relatively good to go

**Comment by Swendvideo** [comment]
For me PR is good, it gives me a new way to look at research, when in past we had to cross our fingers to get something good from 8 free hydro chems, nowadays you have to upgrade clearance to get more passive income, so clearance papers especially from lvl 3-4 will always have a legendary property and it opens a world of relating, It's so good when you getting boosting lvl 3-5 and you create a chem with MST 3 NST 3 and then random cheap property lvl 5 relating it with boosting lvl 5 and done, we have our MST 8 NST 8 chem, so clearance papers is more like random list of high tier property's that I can choose what research chem I want to do today. But I still don't know how the smoke injection works on xenox, so   what I'm trying to ask is what is amount of chem that injecting in xeno? Is the OD of chem is amount how much chem will be in xeno? Or OD is doing nothing in this situation and xeno will take as much as contains in smoke? Also I heard that some healing or Stimulating properties working on xenox, the main question is do Biocidic works on them? This will be logical if they immune to corrosive but what about biocidic?

**Comment by Swendvideo** [comment]
So property's works on xenox this is GREAT. BUT imagine you are solo OT, Mortar ask you for shells, Sadar for full cap rocket and then researchers falls form upper level and says that you now have no free will and you need to create chem nades, mortar asks where shells and sadar cries in corner. Leave the poor OT alone let him do his own stuff, and even times where there no OT and all MT deployed because bravo don't want to secure FOB, researchers have nothing to do but just continue doing Med chems, why we won't add ability to researchers create SMOKE nades NOT EXPLOSIVE, only smoke, the main difference between OT nades is that we will not have ability to insert lighter in them, so any explo chems contain in nade won't work, leave all explosive stuff to OT, All we need a little 180u smoke nade that we can fill 30u potassium 30u phosphorus 30u sugar and 90u research chem

**Comment by stanalbatross** [comment]
TM on saturday

**Comment by Dueu** [comment]
So, I know that there is an idea about "multiple departments working along side to create things in order to change the tides of the battle" thing, the idea in theory sounds good but the problem is that the OT is always busy and even more busy if research managed to create a more explosive and easier to produce chemical to create explosives of mass destruction at terrorific speeds. And honestly I prefer focusing in ordenance research rather than create this weird smoke or foam explosive that the OT didn't test or see if they works so even experience OTs might not even know how to make them properly.

So I got a few ideas to solve this little issue:

Premade Foam Grenades: A grenade that creates 2/3/4 tiles range (3, 5, 7 tiles range in diameter) that can be loaded with a container up to 60 units of content in total (the range of the foam can be ajusted with a screwdriver) and a smaller, UGL compatible foam grenade with 3 tiles range that have a container up to 30 units of content.
This simplifies a bit how the chemical warfare grenades works as if I remember correctly foam (fluorosurfactant mixed with water in 1 to 1 ratio plus the warcrime chemical) applies the on-touch effects of chemicals to mobs and turf and it also is better for marines than smoke as it does not mess their line of sight. Research start with a few of them to do test runs and can be bought through requisition somewhat cheap.

Fire Effects: Make that some effects detrimental for xenos applies those effects through fire but at the cost of damage output, fire size and duration of the flames. Making flamers have the possibility to have more unique effects but reducing its effectiveness to get kills. This could be imbalanced because you could counter the damage reduction with oxidixing or something like that.

**Comment by kiVts** [comment]
alright so after playing like 3 rounds on this tm I can say that, while it fixes a lot of things, it unbalances research even further into oblivion of rng. first to make stim, now not only do you have to be lucky to get one, because now you cant create mst 5 without buying level 3. but you also need to be lucky to get a good drug that is possible to produce and is not poisoning. because you just dont have enough points to create otherwise. 
I should remind that when this pr was created originally a lot of things were still normal, like amplifying negative drug was increasing OD, transmit god dammit was still a thing, now with rounds still being short, you just dont have enough time and points (ios are god blessed though, but they only bring the first intel batch like only after an ig hour. ) to create anything.


this is just hammering the nail into the wound deeper.

**Comment by Dueu** [comment]
So, after 2 rounds of researcher that I got Fire Penetrating trait it turns out it aint that good to begin with.

For making a custom fuel for flamers is bad because balancing Oxidixing/Fueling/Flowing either has too little range, the flame last less than 1 second or its not good enough to beat standard issue fuels. (also no idea how the range of a custom fuel is calculated for a flamer because I end up with a fuel that only has a range of one tile, also this fuel would only benefit the pyro and the few people that got a flamer).

For OT incendiary grenades its good and for rockets and morta shells potentially as it adds also the fire damage upon impact to queen and ravagers but making use of that expecific benefit depends of OT experience. I dont know if mixing the chemical with others would still apply the Fire Penetrating trait.

Conclusion: Pyrotechnic Research its just bad, a plant that naturally produces Explosive 3 chemical or more is orders of magnitudes better than probably the best custom fuel research can make at the moment or just synthesize Explosive chemical made with common ingredients and mass produce it, way simplier and cost-effective and way more destructive.

Still pending to check out MST chemical spread through foam and test the on-touch effect.

**Comment by Swendvideo** [comment]
So I want to talk about Oxidizing touch effect, and it's useless, there's no way that anybody would even test it, for those who don't know Oxidizing can set on fire xenox from lvl 9+, WHY WHYYYYYYY. So to get lvl 9 we NEED clearance 5 because only from lvl 5 we can create lvl 9 and 10 chems, from clearance lvl 3 we can create 8 so some will say "just create lvl 8 and amply 1 time" NO IT'S DOES NOT WORK, YOU NEED TO BUY LVL 5 clearance, so if I remember properly lvl 5 is like 42 point AND YOU ADDITIONALLY NEED TO CREATE LVL 9 chem! As a reminder fueling got touch effect too, it's adds additional time of burn, so if you want to test touch oxi lvl 9 with fueling lvl 3 for example and make 3 common it will cost like 35~ points and with wasted points on clearance lvl 5 we got like 77 points to get touch fire, it's like we need 2 hour round to get it, if you really want to give oxi touch on xenox we need it at least lvl 8+ for clearance lvl 3 that unlocks 1-8 chem lvl

**Comment by Swendvideo** [comment]
So I wanted to test Defib property since it have healing if potency +7, you want to laugh? We can't. Literally. If you try to set Defib lvl synthesis machine will tell you "Max level of defibrillating is 1" like why you even and that Defib heal if +7 if we it can't be higher than 1

**Comment by Unknownity** [comment]
> So I wanted to test Defib property since it have healing if potency +7, you want to laugh? We can't. Literally. If you try to set Defib lvl synthesis machine will tell you "Max level of defibrillating is 1" like why you even add that Defib heal if +7 if we it can't be higher than 1

I looked into it and concluded that it is probably for the best as currently higher levels of defibrillating doesn't do anything and you would be pretty much wasting credits to get that one feature that is in the higher levels.
You can still get high levels of defibrillating through randomly generated chemicals

**Comment by Unknownity** [comment]
> > > So I wanted to test Defib property since it have healing if potency +7, you want to laugh? We can't. Literally. If you try to set Defib lvl synthesis machine will tell you "Max level of defibrillating is 1" like why you even add that Defib heal if +7 if we it can't be higher than 1
> > 
> > 
> > I looked into it and concluded that it is probably for the best as currently higher levels of defibrillating doesn't do anything and you would be pretty much wasting credits to get that one feature that is in the higher levels. You can still get high levels of defibrillating through randomly generated chemicals
> 
> Bro thats literal adding a bug . If you make a feature at least let players make it or remove it . It makes no sense having non obtainable additional effect thats locked on rng

It is not a bug by all means, I won't be adding noob bait unless someone gives a good suggestion of how can defibrillating be better the higher level it is, until then giving you free reign to waste at minimum 6-7 levels worth of credits for a niche feature that is probably outclassed by other healing methods is a waste of credits, potentially ruining your round

**Comment by Diegoflores31** [comment]
> > > > So I wanted to test Defib property since it have healing if potency +7, you want to laugh? We can't. Literally. If you try to set Defib lvl synthesis machine will tell you "Max level of defibrillating is 1" like why you even add that Defib heal if +7 if we it can't be higher than 1
> > > 
> > > 
> > > I looked into it and concluded that it is probably for the best as currently higher levels of defibrillating doesn't do anything and you would be pretty much wasting credits to get that one feature that is in the higher levels. You can still get high levels of defibrillating through randomly generated chemicals
> > 
> > 
> > Bro thats literal adding a bug . If you make a feature at least let players make it or remove it . It makes no sense having non obtainable additional effect thats locked on rng
> 
> It is not a bug by all means, I won't be adding noob bait unless someone gives a good suggestion of how can defibrillating be better the higher level it is, until then giving you free reign to waste at minimum 6-7 levels worth of credits for a niche feature that is probably outclassed by other healing methods is a waste of credits,

> > > > So I wanted to test Defib property since it have healing if potency +7, you want to laugh? We can't. Literally. If you try to set Defib lvl synthesis machine will tell you "Max level of defibrillating is 1" like why you even add that Defib heal if +7 if we it can't be higher than 1
> > > 
> > > 
> > > I looked into it and concluded that it is probably for the best as currently higher levels of defibrillating doesn't do anything and you would be pretty much wasting credits to get that one feature that is in the higher levels. You can still get high levels of defibrillating through randomly generated chemicals
> > 
> > 
> > Bro thats literal adding a bug . If you make a feature at least let players make it or remove it . It makes no sense having non obtainable additional effect thats locked on rng
> 
> It is not a bug by all means, I won't be adding noob bait unless someone gives a good suggestion of how can defibrillating be better the higher level it is, until then giving you free reign to waste at minimum 6-7 levels worth of credits for a niche feature that is probably outclassed by other healing methods is a waste of credits, potentially ruining your round

Since its a kind of expensive trait you could do it so the healing starts at level 2 and increases with potency and if it reaches 7+ then it has the ontouch effect i guess . Also i dont know the if this is related but i have noticed sprays some times inject you with the chemical instead of just aplying the effects is that intended ?

**Comment by Ticker175** [comment]
As someone with extensive experience (300+ hours) with research I think that the current master branch (the normal version) is very enjoyable, it still has some issues but all in all it’s very fun.

Things I really like about this PR are; the income increase is great and research has needed this for a while now, new property’s (well not new, reenabled) and effects are great! Always love new things to test and “research” haha, I also really like the cooperation it encourages with research and OT i think that that kind of department collaboration is awesome.

With all that being said there are of course some negatives I see with this and some things that at least I think should be fixed with research in general.

The OT/FIRE chem property’s are very weak (even with fire pen 10 flowing 10 fueling 10 and oxidising 10 the chem can’t even kill a drone in less than a minute, let alone hurt a rav (even though the fire pen property is supposed to be a good anti rav weapon)) and seem to be a complete waste of time (this goes for research chem fire in general)

We need new chemical property’s for stims, all we have currently is speed (MST), anti stun (NST), healing property’s and the various special healing properties (anti larva, organ stab, bone mend, ect). New property’s like melee/strength boost, durability boost (make someone more ROBUST!), darkvision/nightvision (see in the dark), ability speed increase (make someone preform actions faster eg. Build cades, apply splints, do surgery, remove cuffs, ect), endurance boost (maybe sprint ability or anti disarm/knockdown)

Greenos need to be addressed, I have 2 main concerns/suggestions with greenos. Firstly I extremely extremely HATE how greenos require clearance 5x (more so how many points it costs to get to that), it is just a massive fuck you to anyone wanting to make stims or play with the new changes. My suggestion is to make greenos seperate from stims and instead of clearance change it to cost Tech points or something that requires the researcher to actually work towards getting greenos and not just click a button and read a wiki eg. Assemble a complex chem or get a certain item from the planet or all of the above. Also greenos should be timelocked via an in character method like authorisation from high command or support from W-Y (maybe have a W-Y “response team” of scientists that bring something to enable the researchers to make their own hive (I especially like this because it allows people that die early on to come back and adds the possibility of more side plot))
Just please add something to stop the bald ass researchers that clearance rush EVERY ROUND from fucking people who want to make stims.


All in all love new things but please keep in mind the already implemented features that have issues

Thanks for reading!  😉😁

**Comment by CodingIsNotEasy** [comment]
Issue one: ever since this was introduced, there's been a flocking in from OT players into research. OT players usually use up all their stuff pretty quickly so what they'll do is, they're play researcher, set turing to permanently produce polynitric acid then they'll go continue their round as ghetto OT while using research dispensers to make ammonia/methane and i hate it. as a researcher that wants some mutagen for their trays, there's always that one OT wannabe researcher that drains both turing and dispenser. they'll buy clearance with points they did nothing to earn and get pissy if we get MST, NST or queen and egg. ONE GUY LITERALLY THREW AWAY ALL THE MONKEY CUBES on three different occasions cause he wanted to save the points for some high level of EXP or OXI chem even though we had both queen and egg. and lord help me if they found out we got Fire Pen. they'll just lose their shit.
on numerous occasions, i had OTs straight up hack my lab doors and go in to see what properties i got.
some separation needs to happen. make OT chems/properties non attainable via dispensers or something. and remove fire-penetrating.

Issue Two: Research almost NEVER gets a stim down on the ground due to point waste on needless clearance or an intention to increase passive research gain thus leaving no points to make anything worth getting down on the ground or even anything with a tangible effect and more than 5u OD. then you added to that tying property level to clearance level and just completely shat on any chance research had of making a stim in a round shorter than 3 hours. rounds rarely break the hour mark these days.

Issue Three: no uses ACID harness. literally no one. i can count on two fingers the number of people i've ever seen use/request them.

TL;DR, Separation of OT from Research is needed. remove Firepen along with EXP, OXI and Fuelling in addition to TOX4 effect on weeds. a Remove caps on property levels. No one cares about the ACID harness

**Comment by Ticker175** [comment]
So I agree that research should be focused on research but I don’t see a problem with (AFTER the first round of muta for botany obviously) for example having the turning automatically making methane or poly in the background whilst your working on botany and chems, it’s automatic to bottle to the fridge so it’s just free chems (helping others is always nice, and if you help them they will be much more likely to help you if you need para or formaldehyde later on). A suggestion for dealing with the bootleg OT researchers, ask them to just wait till after you get the muta for botany, and explain why they should wait. There is no point screaming and yelling and being a dick, there are enough assholes online just be nice 👍. 
Admin help is also available for stuff like people maliciously destroying stuff or spending points

I don’t know a single doctor, nurse, researcher, cmo or anyone that uses a chem dispenser that doesn’t believe that the power generation for the chem dispensers desperately needs to be improved.
Along with this OT also desperately needs some way to get more chems themselves because it’s EXTREMELY rare that you will have someone from medical that is willing to give you more than a bucket or 2 of chems.

regarding lab doors I agree that getting your doors hacked is a absolute fucking pain. If I had my way I would make research doors (the main 2) hack proof like the req doors

Yeah I’ve already said my opinion on all the fire properties. They need to be buffed, reworked or removed

I agree, the clearance level gates for property levels is just terrible and further discourages stim creation on top of the increased cost that was implemented by this TM.

I don’t see a problem with acid harnesses and removing them would be stupid and pointless, they arnt doing any harm so leave them alone. However I do believe that they would be used more if they accepted larger containers and not just vials. I would change them to accept normal 60 unit beakers and possibly large beakers too but that should probably be balance tested as it might be a little unbalanced

**Comment by Dueu** [comment]
### My rant about research
The era of research being able to make monstrous amounts of ultrazine and mix it with anti addictive chemical and make an injector that turns marines into high speed roid rage trains with guns is gone, early stimulant is gone too if you are lucky enough to get MST and NST as you are limited to buy clearence to create something in level 6 or 8, Pyrotechnic research is bad and the only thing BESIDE greenos that is not that nerfed is the Explosive property being a beast even in low potency like 3.

**A few opinions I thought about reading the content of the PR**

1. The high level requirement of the Oxidixing property to apply the effect on_touch to set on fire mobs makes it not worth the effort to even get it and the few ways you can apply the effect makes it even less usable.
2. I'm not a fan of having to depend on the OT to make anti xeno foam with chemicals harmful to them like corrosive as its a posibility to not have an OT or the OT give priority explosives and deny your foam grenades production, in the absence of the OT your could ask a MT but you know MT ain't that trained in making ordenance and less in foam grenades. Also chemical sprayers are just bad in that expecific task, on non combat instance yes I will use them to deliver speed stimulants but thats it.
3. I know the reason why not super chemical sprayers or syringe guns, the effects can potentially be too intense and let a select few to do insane things if the chemical is strong enough. Basically making an extra spec that is specialized on chemical warfare also the effects could stack with other things like combine slug slowdown with the slowdown of a chemical reducing to a crawl a xeno. (who knows maybe its actually a good idea have a very niche spec that works along side research)

**A possible problem with the Corrosive property**

I understand that Corrosive is a double edge sword property that can injure both ungas and left very vulnerable usually extremely armored xenos like the crusher but it destroys headgear and items on the floor which could potentially melt a spec gun and I consider that far too much risk, it makes sense but gameplay wise is just too much. Could be hilarious if you are about to throw the corrosive grenade then queen screeches, about a dozen people drop their guns, the nade goes off and the foam left a dozen marines gunless.

**Some aclarations and extra small notes**

This rant is base on the information of the gitlab and this PR with some small experiences in game but probably everything is in some form in hypothetical scenarios so don't take my word as 100% real.

Generally I would like regardless of the changes this PR to get merged and it brings needed new content, legendary properties are now somewhat obtainable rather than this 1 in a 50 rounds thing that doesn't matter because either side won too fast to make a difference.

This PR I think it makes easier to greeno rush as you need less points to buy clearence, so yeah probably greeno rushers researchers will become a thing as its not nerfed and the components needed are not controlled by RNG and limitations rather how much xenos guard corpses and eggs during combat, what they evolve into (could be a chance to have no praetorians and carriers in the hive) and how likely are they to die.

I made this rant to express myself a bit of how research is now, so if is there any negative emotions into the rant my sincere apologies.

**Comment by kiVts** [comment]
> > > > So I wanted to test Defib property since it have healing if potency +7, you want to laugh? We can't. Literally. If you try to set Defib lvl synthesis machine will tell you "Max level of defibrillating is 1" like why you even add that Defib heal if +7 if we it can't be higher than 1
> > > 
> > > 
> > > I looked into it and concluded that it is probably for the best as currently higher levels of defibrillating doesn't do anything and you would be pretty much wasting credits to get that one feature that is in the higher levels. You can still get high levels of defibrillating through randomly generated chemicals
> > 
> > 
> > Bro thats literal adding a bug . If you make a feature at least let players make it or remove it . It makes no sense having non obtainable additional effect thats locked on rng
> 
> It is not a bug by all means, I won't be adding noob bait unless someone gives a good suggestion of how can defibrillating be better the higher level it is, until then giving you free reign to waste at minimum 6-7 levels worth of credits for a niche feature that is probably outclassed by other healing methods is a waste of credits, potentially ruining your round

 X potency blocks suffocating damage, does not heal
or

and also, does this heal when your alive like bica or during defibrillating like normal defib, if first, second seems a better idea, so X*2 potency, enabled by X potency and beyond heals when your getting defibed by drug like real defib, does not deal heart damage(so maximum heal would be 18 on level 9 and 10 on level 5)

**Comment by Weuyn** [comment]
higher levels of defibrilating allows you to relate it onto different levels to get rid of a harmful property. Level 1 properties are actually pretty rare.

**Comment by Dueu** [comment]
It turns out fluorosurfactant doesn't apply the on_touch effects like the chem smoke does for some code reasons, I made an Issue about it in detail: https://github.com/cmss13-devs/cmss13/issues/932#issue-1348307245

**Comment by Unknownity** [comment]
> It turns out fluorosurfactant doesn't apply the on_touch effects like the chem smoke does for some code reasons, I made an Issue about it in detail: [#932 (comment)](https://github.com/cmss13-devs/cmss13/issues/932#issue-1348307245)

Might be an oversight that ben forgot to include in since there's nothing about it in the code, ill uh look into it more

File: code/datums/components/healing_reduction.dm
Comment (stanalbatross) [inline code/datums/components/healing_reduction.dm:70]: why parse mob as an arg when it's not used 🤔

File: code/datums/components/healing_reduction.dm
Comment (stanalbatross) [inline code/datums/components/healing_reduction.dm:74]: same here

File: code/modules/reagents/chemistry_properties/prop_negative.dm
Comment (stanalbatross) [inline code/modules/reagents/chemistry_properties/prop_negative.dm:58]: what is this istype madness 💀

File: code/modules/reagents/chemistry_properties/prop_negative.dm
Comment (stanalbatross) [inline code/modules/reagents/chemistry_properties/prop_negative.dm:163]:
```suggestion
			to_chat(M, SPAN_WARNING("\The [O] melts."))
```

**Review by stanalbatross — CHANGES_REQUESTED** [review]

File: code/modules/reagents/chemistry_properties/prop_negative.dm
Comment (Unknownity) [reply code/modules/reagents/chemistry_properties/prop_negative.dm:58]: code that mainly was transferred from what we already have, see code/modules/reagents/chemistry_reagents/toxin.dm 166-193
the only change is that its now applied to a property instead of a specific chemical

**Review by stanalbatross — APPROVED** [review]
approved, if there are bugs we can fix them later

**Comment by Dueu** [comment]
Research chemical with the property Oxidizing 8 didn't set on fire by itself. Tested using sprayers, splashing with containers and smoke on a human target (myself), unless the fire is so short live it doesn't set on fire but I don't know about that.

**Pull request merged**
